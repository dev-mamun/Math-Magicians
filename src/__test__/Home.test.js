/** ****************************************
 * Project: math-magicians
 * File: Home.test.js
 * Created: 4/8/23
 * Author: Abdullah Al Mamun <mamun1214@gmail.com>
 ****************************************** */
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import Home from '../pages/Home';

it('Home Render Correctly', () => {
  const tree = render(<Home />);
  expect(tree).toMatchSnapshot();
});

it('Home UI', () => {
  render(
    <BrowserRouter>
      <Home />
    </BrowserRouter>,
  );
  const Element = screen.getByText(/Welcome To Our Page!/i);
  expect(Element).toBeInTheDocument();
});
