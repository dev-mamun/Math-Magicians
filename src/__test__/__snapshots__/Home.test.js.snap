// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Render Correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2>
          Welcome To Our Page! 
        </h2>
        <p>
          We are delighted to have you here and share our passion for learning and inspiration with you. As you browse our site, you will find a variety of resources and tools to help you achieve your goals and unlock your potential. Our calculator is designed to make complex math problems a breeze, while our randomly generated quotes can spark your creativity and motivate you to reach for the stars. At the heart of our website is a commitment to accessibility and inclusivity. We believe that everyone should have access to the knowledge and resources they need to succeed, and we are proud to offer our content and tools to a global audience.
        </p>
        <br />
        <p>
          We are also committed to building a supportive and engaged community, and we welcome your feedback and ideas. Whether you're a student, professional, or lifelong learner, we hope you'll find something here that inspires and empowers you.Thank you for visiting our website, and we look forward Lorem, ipsum dolor sit amet consectetur adipisicing elit. Animi dolorem, maxime nemo doloremque nihil assumenda labore omnis nisi, fugiat autem libero quae ullam veritatis Lorem, ipsum dolor sit amet consectetur adipisicing elit. Animi dolorem, maxime nemo doloremque nihil assumenda labore omnis nisi, fugiat autem libero quae ullam veritatis Lorem, ipsum dolor sit amet consectetur adipisicing elit. Animi dolorem, maxime nemo doloremque nihil assumenda labore omnis nisi, fugiat autem libero quae ullam veritatis blanditiis.Consequuntur nostrum natus numquam sunt. blanditiis. Consequuntur nostrum natus numquam sunt. blanditiis. Consequuntur nostrum natus numquam sunt.to connecting with you as we continue to grow and evolve.
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2>
        Welcome To Our Page! 
      </h2>
      <p>
        We are delighted to have you here and share our passion for learning and inspiration with you. As you browse our site, you will find a variety of resources and tools to help you achieve your goals and unlock your potential. Our calculator is designed to make complex math problems a breeze, while our randomly generated quotes can spark your creativity and motivate you to reach for the stars. At the heart of our website is a commitment to accessibility and inclusivity. We believe that everyone should have access to the knowledge and resources they need to succeed, and we are proud to offer our content and tools to a global audience.
      </p>
      <br />
      <p>
        We are also committed to building a supportive and engaged community, and we welcome your feedback and ideas. Whether you're a student, professional, or lifelong learner, we hope you'll find something here that inspires and empowers you.Thank you for visiting our website, and we look forward Lorem, ipsum dolor sit amet consectetur adipisicing elit. Animi dolorem, maxime nemo doloremque nihil assumenda labore omnis nisi, fugiat autem libero quae ullam veritatis Lorem, ipsum dolor sit amet consectetur adipisicing elit. Animi dolorem, maxime nemo doloremque nihil assumenda labore omnis nisi, fugiat autem libero quae ullam veritatis Lorem, ipsum dolor sit amet consectetur adipisicing elit. Animi dolorem, maxime nemo doloremque nihil assumenda labore omnis nisi, fugiat autem libero quae ullam veritatis blanditiis.Consequuntur nostrum natus numquam sunt. blanditiis. Consequuntur nostrum natus numquam sunt. blanditiis. Consequuntur nostrum natus numquam sunt.to connecting with you as we continue to grow and evolve.
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
